homeassistant/components/sensor/nilu_air_quality.py

```diff
@@ -0,0 +1,238 @@
+"""
+Sensor for checking the air quality around Norway.
+
+Data delivered by luftkvalitet.info and nilu.no.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.nilu_air_quality/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    ATTR_ATTRIBUTION, CONF_LATITUDE, CONF_LONGITUDE, CONF_MONITORED_CONDITIONS,
+    CONF_NAME, CONF_SHOW_ON_MAP, CONF_UNIT_OF_MEASUREMENT)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['niluclient==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=30)
+
+CONF_AREA = 'area'
+CONF_STATION = 'station'
+DEFAULT_NAME = 'NILU'
+
+CONF_ATTRIBUTION = "Data provided by luftkvalitet.info and nilu.no"
+ATTR_POLLUTION_INDEX = "pollution_index"
+ATTR_MAX_POLLUTION_INDEX = 'max_pollution_index'
+
+CONF_ALLOWED_AREAS = [
+    'Ålesund',
+    'Zeppelinfjellet',
+    'Tustervatn',
+    'Trondheim',
+    'Tromsø',
+    'Sør-Varanger',
+    'Stavanger',
+    'Sarpsborg',
+    'Sandve',
+    'Prestebakke',
+    'Oslo',
+    'Narvik',
+    'Moss',
+    'Mo i Rana',
+    'Lørenskog',
+    'Lillestrøm',
+    'Lillesand',
+    'Lillehammer',
+    'Kårvatn',
+    'Kristiansand',
+    'Karasjok',
+    'Hurdal',
+    'Harstad',
+    'Hamar',
+    'Halden',
+    'Grenland',
+    'Gjøvik',
+    'Fredrikstad',
+    'Elverum',
+    'Drammen',
+    'Bærum',
+    'Brumunddal',
+    'Bodø',
+    'Birkenes',
+    'Bergen'
+]
+
+CONF_ALLOWED_MEASURABLE_COMPONENTS = [
+    'index',
+    'PM1',
+    'PM10',
+    'PM2.5',
+    'NO',
+    'NO2',
+    'NOx',
+    'O3',
+    'CO',
+    'SO2'
+]
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Inclusive(CONF_LATITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.latitude,
+    vol.Inclusive(CONF_LONGITUDE, 'coordinates',
+                  'Latitude and longitude must exist together'): cv.longitude,
+    vol.Required(CONF_MONITORED_CONDITIONS):
+        vol.All(cv.ensure_list, [vol.In(CONF_ALLOWED_MEASURABLE_COMPONENTS)]),
+    vol.Exclusive(CONF_AREA, 'station_collection',
+                  'Can only configure one specific station or '
+                  'stations in a specific area pr sensor. '
+                  'Please only configure station or area.'):
+        vol.All(cv.string, vol.In(CONF_ALLOWED_AREAS)),
+    vol.Exclusive(CONF_STATION, 'station_collection',
+                  'Can only configure one specific station or '
+                  'stations in a specific area pr sensor. '
+                  'Please only configure station or area.'):
+        cv.string,
```

Comment:
Why not allow a list of stations?

Reply:
Thanks for the code review! 
I added it as a string to align with how the `area` input works, which is the same type of input. Originally I had `areas` as a list, but it was suggested in code review that it might work better if it was one sensor for each config entry. I transferred this thinking to the `station` input as well, to not have too many approaches to the config on one sensor platform.

Reply:
It seemed natural to me to be able to specify a list of stations, since the `location_client` will return a list of stations and we already iterate a list of stations. But other than that I don't have an opinion.

Reply:
The `cv.ensurelist` would support both our cases, so it wouldn't change much more than having an extra 's' on the end of the config name. I also can agree that its more natural within the code to have the list. 
If you guys think that it fits better to have a list there, I would gladly change it to a list 😃
Sorry for the delayed response, btw.

Reply:
Sounds good with `cv.ensure_list`.